Update pm2 commands' page - Closes 1705 (#1750)
LiskHQ/lisk-service: docs/antora/modules/ROOT/pages/management/source.adoc

cd ./tests && rm -rf node_modules
----
====

=== Install npm dependencies

[source,bash]
----
make build-local
----

=== Reset the Lisk Service database

To reset the database of Lisk Service, drop the respective MySQL and Redis databases.

==== Drop the MySQL database

. Stop Lisk Service
+
[source,bash]
----
npm stop
----

. Login to MySQL with the `lisk` user.
+
[source,bash]
----
mysql -u lisk -ppassword
----

. Drop the database.
+
[source,bash]
----
mysql> drop database lisk;
----

. Create a fresh database.
+
[source,bash]
----
mysql> create database lisk;
----

. Quit MySQL.
+
[source,bash]
----
mysql> quit;
----

NOTE: The process to re-index all the data commences after Lisk Service starts again following a database reset. The process can be rather time-consuming and could take several hours, depending on the length of the chain.

==== Flush Redis DB

Reset the databases for Redis after dropping the MySQL database:

[source,bash]
----
redis-cli flushall
----

[NOTE]
====
The `flushall` command truncates all existing Redis databases:

> Deletes all the keys of all the existing databases, not just the current selected one. This command never fails.

> For more information check the Redis documentation: {url_FLUSHALL}[FLUSHALL]

Reviewer: ```suggestion
For more information, check the Redis documentation: {url_FLUSHALL}[FLUSHALL]
```

= PM2 commands
Mona Bärenfänger <mona@lightcurve.io> Muhammad Talha <muhammad.talha@lightcurve.io>
:description: Describes how to manage Lisk Service with PM2.
:toc:
:idseparator: -
:idprefix:
:imagesdir: ../assets/images

// External URLs
:url_pm2_docs: https://pm2.keymetrics.io/docs/usage/quick-start/
:url_FLUSHALL: https://redis.io/commands/FLUSHALL
:url_mainchain_client: https://github.com/LiskHQ/lisk-core#example-using-pm2

:url_global_cli: build-blockchain/create-blockchain-client.adoc#using-the-client-cli-globally
:url_build_hello_client: build-blockchain/index.adoc#the-hello-world-client
:url_core_client: run-blockchain/index.adoc#how-to-set-up-a-lisk-mainnet-node

> On this page, you'll find the most relevant PM2 commands for running source-based Lisk Service installation.

Reviewer 2: ```suggestion
On this page, you'll find the most relevant PM2 commands for running a source-based Lisk Service installation.
```

For more details about PM2, see {url_pm2_docs}[PM2 docs].

> Moreover, We'll also talk about the steps to reset a Lisk Service node.

Reviewer: ```suggestion
Moreover, we'll also talk about the steps to reset a Lisk Service node.
```

│ 3   │ lisk-service-blockchain-indexer         │ default     │ 0.7.0-… │ fork    │ 33338    │ 0s     │ 16   │ online    │ 0%       │ 54.3mb   │ XYZ      │ disabled │
│ 8   │ lisk-service-export                     │ default     │ 0.7.0-… │ fork    │ 33355    │ 0s     │ 15   │ online    │ 0%       │ 14.1mb   │ XYZ      │ disabled │
│ 5   │ lisk-service-fee-estimator              │ default     │ 0.7.0-… │ fork    │ 33346    │ 0s     │ 15   │ online    │ 0%       │ 46.9mb   │ XYZ      │ disabled │
│ 0   │ lisk-service-gateway                    │ default     │ 0.7.0-… │ fork    │ 33326    │ 0s     │ 15   │ online    │ 0%       │ 65.6mb   │ XYZ      │ disabled │
│ 7   │ lisk-service-market                     │ default     │ 0.7.0-… │ fork    │ 33351    │ 0s     │ 15   │ online    │ 0%       │ 36.9mb   │ XYZ      │ disabled │
│ 6   │ lisk-service-transaction-statistics     │ default     │ 0.7.0-… │ fork    │ 33348    │ 0s     │ 15   │ online    │ 0%       │ 43.5mb   │ XYZ      │ disabled │
└----------------------------------------------------------------------------------------------------------------------------------------------------------------------------┘
----
====

=== Logs

> To check the logs for the different microservices of Lisk Service, use the command `pm2 logs PROCESS_NAME`, where `PROCESS_NAME` is the respective PM2 managed process that contains the logs you wish to view.

Reviewer: Please feel free to edit the recommendation.
```suggestion
To check the logs for the different microservices of Lisk Service, use the command `pm2 logs PROCESS_NAME`, where `PROCESS_NAME` is the respective PM2 managed process that contains the logs you wish to view.
Alternatively, it is also possible to the PM2 process ID as seen above. For example, `pm2 logs 0` to check the Lisk Service Gateway logs.
```

Reviewer 2: For the recommendation, I would just suggest the following :
"Alternatively, it is also possible to check the PM2 process ID as shown above. For example, `pm2 logs 0` to check the Lisk Service Gateway logs".

Author: ```suggestion
To check the logs for the different microservices of Lisk Service, use the command `pm2 logs PROCESS_NAME`, where `PROCESS_NAME` is the respective PM2-managed process that contains the logs you wish to view.
Alternatively, it is also possible to check the logs for a specific PM2 process based on its process ID. For example, `pm2 logs 0` can be used to check the Lisk Service Gateway logs.
```

> For example, to see the logs for the Gateway microservice, execute:

Reviewer 2: ```suggestion
For example, to see the logs for the Gateway microservice, execute the following command:
```

0|lisk-ser | 2023-07-19 17:53:09 016: 2023-07-19T17:53:09.016 INFO [TEMP_SERVICE_GATEWAY] Waiting for service(s) 'indexer, connector'...
0|lisk-ser | 2023-07-19 17:53:09 018: 2023-07-19T17:53:09.018 INFO [REGISTRY] '$node' service is registered.
0|lisk-ser | 2023-07-19 17:53:09 018: 2023-07-19T17:53:09.018 INFO [$NODE] Service '$node' started.
0|lisk-ser | 2023-07-19 17:53:10 225: 2023-07-19T17:53:10.223 INFO [REGISTRY] Node 'XYZ.local-74790' connected.
0|lisk-ser | 2023-07-19 17:53:10 274: 2023-07-19T17:53:10.274 INFO [REGISTRY] Node 'XYZ.local-74787' connected.
0|lisk-ser | 2023-07-19 17:53:10 325: 2023-07-19T17:53:10.325 INFO [REGISTRY] Node 'XYZ.local-74798' connected.
0|lisk-ser | 2023-07-19 17:53:10 346: 2023-07-19T17:53:10.345 INFO [REGISTRY] Node 'XYZ.local-74772' connected.
0|lisk-ser | 2023-07-19 17:53:10 415: 2023-07-19T17:53:10.414 INFO [REGISTRY] Node 'XYZ.local-74804' connected.
0|lisk-ser | 2023-07-19 17:53:10 485: 2023-07-19T17:53:10.484 INFO [REGISTRY] Node 'XYZ.local-74795' connected.
0|lisk-ser | 2023-07-19 17:53:10 676: 2023-07-19T17:53:10.675 INFO [REGISTRY] Node 'XYZ.local-74779' connected.
----
====

=== Flush

> Remove all the logs of PM2 managed processes.

Reviewer: ```suggestion
Remove all the logs of PM2 managed processes, defined in the specified config file.
```

Reviewer 2: I would just suggest a minor update to the following sentence:
"Remove all the logs of the PM2 managed processes defined in the specified config file".

Author: ```suggestion
Remove all the logs of the PM2-managed processes defined in the specified config file.
```

[PM2] [lisk-service-blockchain-connector](2) ✓
[PM2] [lisk-service-blockchain-coordinator](4) ✓
[PM2] [lisk-service-fee-estimator](5) ✓
[PM2] [lisk-service-transaction-statistics](6) ✓
[PM2] [lisk-service-market](7) ✓
[PM2] [lisk-service-export](8) ✓
┌----------------------------------------------------------------------------------------------------------------------------------------------┐
│ id  │ name      │ namespace   │ version │ mode    │ pid      │ uptime │ ↺    │ status    │ cpu      │ mem      │ user     │ watching │
└----------------------------------------------------------------------------------------------------------------------------------------------┘
[PM2][WARN] Current process list is not synchronized with saved list. App lisk-service-gateway lisk-service-blockchain-app-registry lisk-service-blockchain-connector lisk-service-blockchain-indexer lisk-service-blockchain-coordinator lisk-service-fee-estimator lisk-service-transaction-statistics lisk-service-market lisk-service-export differs. Type 'pm2 save' to synchronize.
----
====

== Resetting Lisk Service

=== Clean all run-time files with dependencies

[source,bash]
----

> make clean

Reviewer: Since, we're discussing only about the pm2 managed processes, lets specify how to clean the locally installed dependencies with `make clean-local`. `make clean` will additionally remove the docker images, if they exist, which is unintended.
```suggestion
make clean-local
```

Untagged: lisk/service_blockchain_indexer:latest
Deleted: sha256:a3640ec100f7489c0b14043bec2f09e7f81cbe7bd3f044b9b12d04bc6976dda1
Untagged: lisk/service_blockchain_coordinator:latest
Deleted: sha256:a4b8aa984dba9028f60ff4a2194924065989a1aaa8429eb93119cf5036700dac
Untagged: lisk/service_transaction_statistics:latest
Deleted: sha256:2fbb60090859f858a87595af6c82ce3cf07015747b937bd322749efd3dfd6f9e
Untagged: lisk/service_fee_estimator:latest
Deleted: sha256:0759e75bde038460c42eccc3979c6c851824f7306e2c295608a58606b4de54d4
Untagged: lisk/service_market:latest
Deleted: sha256:0bfe5f3dbcabbe40b38718444dc5fe578e5faab35b0c53504054924190c3f928
Untagged: lisk/service_export:latest
Deleted: sha256:c4b7f4023da7e81271f20fd05afc3f596074d013d39c6cd0acd0f1409e854ec1
Error response from daemon: No such image: lisk/service_template:latest
Error response from daemon: No such image: lisk/service_tests:latest
----
====

=== Install npm dependencies

[source,bash]
----
make build-local
----

=== Reset the Lisk Service database

To reset the database of Lisk Service, drop the respective MySQL and Redis databases.

==== Drop the MySQL database

. Stop Lisk Service
+
[source,bash]
----
npm stop
----

> . Login to MySQL with the lisk user

Reviewer 2: ```suggestion
. Login to MySQL with the Lisk user
```

Author: ```suggestion
. Login to MySQL with the `lisk` user
```

+
[source,bash]
----
mysql -u lisk -ppassword
----

. Drop the database
+
[source,bash]
----
mysql> drop database lisk;
----

. Create fresh database
+
[source,bash]
----
mysql> create database lisk;
----

. Quit MySQL
+
[source,bash]
----
mysql> quit;
----

> NOTE: When Lisk Service is started again after a database reset, then the process to reindex all the data is initiated. This can be quite time-consuming and may take up to 2 hours.

Reviewer: ```suggestion
NOTE: When Lisk Service is started again after a database reset, then the process to re-index all the data is initiated. This can be quite time-consuming and may take up hours, depending on the length of the chain.
```

Reviewer 2: imho, I would suggest just slightly rewording this sentence to the following: 
"The process can be rather time-consuming and could take several hours, depending on the length of the chain."

Author: ```suggestion
NOTE: The process to re-index all the data commences after Lisk Service starts again following a database reset. The process can be rather time-consuming and could take several hours, depending on the length of the chain.
```

==== Flush Redis DB

Reset the databases for Redis after dropping the MySQL database:

[source,bash]
----
redis-cli flushall
----

[NOTE]
====

> The `flushall` command removes all existing Redis databases:

Reviewer: ```suggestion
The `flushall` command truncates all existing Redis databases:
```
or
```suggestion
The `flushall` command removes all available entries from all the existing Redis databases:
```

[PM2] [lisk-service-fee-estimator](5) ✓
[PM2] [lisk-service-transaction-statistics](6) ✓
[PM2] [lisk-service-market](7) ✓
[PM2] [lisk-service-export](8) ✓
┌----------------------------------------------------------------------------------------------------------------------------------------------┐
│ id  │ name      │ namespace   │ version │ mode    │ pid      │ uptime │ ↺    │ status    │ cpu      │ mem      │ user     │ watching │
└----------------------------------------------------------------------------------------------------------------------------------------------┘
[PM2][WARN] Current process list is not synchronized with saved list. App lisk-service-gateway lisk-service-blockchain-app-registry lisk-service-blockchain-connector lisk-service-blockchain-indexer lisk-service-blockchain-coordinator lisk-service-fee-estimator lisk-service-transaction-statistics lisk-service-market lisk-service-export differs. Type 'pm2 save' to synchronize.
----
====

== Resetting Lisk Service

=== Clean all run-time files with dependencies

[source,bash]
----
make clean-local
----

.Response
[%collapsible]
====
.Example output
[source,bash]
----

> rm -rf node_modules
> cd ./framework && rm -rf node_modules
> cd ./services/blockchain-app-registry && rm -rf node_modules
> cd ./services/blockchain-connector && rm -rf node_modules
> cd ./services/blockchain-coordinator && rm -rf node_modules
> cd ./services/blockchain-indexer && rm -rf node_modules
> cd ./services/transaction-statistics && rm -rf node_modules
> cd ./services/fee-estimator && rm -rf node_modules
> cd ./services/market && rm -rf node_modules
> cd ./services/gateway && rm -rf node_modules
> cd ./services/export && rm -rf node_modules
> cd ./services/template && rm -rf node_modules
> cd ./tests && rm -rf node_modules
> docker rmi lisk/service_gateway \
> 	lisk/service_blockchain_app_registry \
> 	lisk/service_blockchain_connector \
> 	lisk/service_blockchain_indexer \
> 	lisk/service_blockchain_coordinator \
> 	lisk/service_transaction_statistics \
> 	lisk/service_fee_estimator \
> 	lisk/service_market \
> 	lisk/service_export \
> 	lisk/service_template \
> 	lisk/service_tests; :
> Untagged: lisk/service_gateway:latest
> Deleted: sha256:5749869435bc9c562af29435999a4e4afee6fa0a7cb8cb188ed87805b9b793ba
> Untagged: lisk/service_blockchain_app_registry:latest
> Deleted: sha256:3dbea0971611d4072e06bc677daf53611696312c7ca3c2bdf5f3854fd5310de2
> Untagged: lisk/service_blockchain_connector:latest
> Deleted: sha256:c88ea0780275c54b944f524ee85b6d935d155261bf246fd9dd7331ea0a3a0ae6
> Untagged: lisk/service_blockchain_indexer:latest
> Deleted: sha256:a3640ec100f7489c0b14043bec2f09e7f81cbe7bd3f044b9b12d04bc6976dda1
> Untagged: lisk/service_blockchain_coordinator:latest
> Deleted: sha256:a4b8aa984dba9028f60ff4a2194924065989a1aaa8429eb93119cf5036700dac
> Untagged: lisk/service_transaction_statistics:latest
> Deleted: sha256:2fbb60090859f858a87595af6c82ce3cf07015747b937bd322749efd3dfd6f9e
> Untagged: lisk/service_fee_estimator:latest
> Deleted: sha256:0759e75bde038460c42eccc3979c6c851824f7306e2c295608a58606b4de54d4
> Untagged: lisk/service_market:latest
> Deleted: sha256:0bfe5f3dbcabbe40b38718444dc5fe578e5faab35b0c53504054924190c3f928
> Untagged: lisk/service_export:latest
> Deleted: sha256:c4b7f4023da7e81271f20fd05afc3f596074d013d39c6cd0acd0f1409e854ec1
> Error response from daemon: No such image: lisk/service_template:latest
> Error response from daemon: No such image: lisk/service_tests:latest

Author: ```suggestion
rm -rf node_modules
cd ./framework && rm -rf node_modules
cd ./services/blockchain-app-registry && rm -rf node_modules
cd ./services/blockchain-connector && rm -rf node_modules
cd ./services/blockchain-coordinator && rm -rf node_modules
cd ./services/blockchain-indexer && rm -rf node_modules
cd ./services/transaction-statistics && rm -rf node_modules
cd ./services/fee-estimator && rm -rf node_modules
cd ./services/market && rm -rf node_modules
cd ./services/gateway && rm -rf node_modules
cd ./services/export && rm -rf node_modules
cd ./services/template && rm -rf node_modules
cd ./tests && rm -rf node_modules
```
